server/http/httpclient/httpclient.go

```diff
@@ -0,0 +1,119 @@
+package httpclient
+
+import (
+	"errors"
+	"flag"
+	"net"
+	"net/http"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/server/metrics"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/net/publicsuffix"
+)
+
+// Tests often need to make HTTP requests to localhost -- set this flag to permit those requests.
+var allowLocalhost = flag.Bool("httpclient.allow_localhost", false, "Allow HTTP requests to localhost")
```

Comment:
nit: maybe `http.client` so it mirrors the structure of the `cache.client` flags (from cachetools.go)

```suggestion
var allowLocalhost = flag.Bool("http.client.allow_localhost", false, "Allow HTTP requests to localhost")
```